ios/FluentUI/Navigation/NavigationBar.swift

```diff
@@ -589,6 +591,12 @@ open class NavigationBar: UINavigationBar, TokenizedControlInternal, TwoLineTitl
         titleView.avatarOverrideStyle = style
     }
 
+    /// Override BadgeLabelStyle for navigation bar
+    /// - Parameter badgeLabelStyle: updated style to be used
+    @objc public func overrideBadgeLabelStyle(_ badgeLabelStyle: Style) {
```

Comment:
This API is unclear and relies on very specific behaviors about `Style`. That class describes the entire notification bar, not just the label style, so we are making two assumptions:
1) Callers of this API will understand exactly what will happen to the badge label if they specify a different nav bar style.
2) We will never change what label styles are associated with each navigation bar style.

If we are going to provide a way to change the color of the label, we should expose the final outcome of this: the button's `BadgeLabelStyle`, and have people override that.

Reply:
only issue with exposing `BadgeLabelStyle` is that we can't keep as null by default since @Objc don't allow this with optional in swift. So, we need to assign some default value which means even when user not setting this explicitly it will be using that value.

Purpose of this API is that user explicitly call this to set, else it will be null.

Please let me know if we can achieve null handling in some better way? Basically we should have it null by default.

Reply:
making that property optional direct not allowed as it seems optional Enum in swift with property marked as @Objc not allowed. My thought process here is to use this property only when not nil. Assigning default value will create some confusion.
<img width="1216" alt="Screenshot 2024-02-29 at 11 56 23 AM" src="https://github.com/microsoft/fluentui-apple/assets/102029000/7a9d731b-a9d1-42b7-a2dc-949cfab008b3">

Reply:
@mischreiber Have used wrapper class to fix this, please have a look.

Reply:
We definitely don't want to have to create a wrapper class just to expose to objC. One question first - do we actually need to consume this from objC? Assuming we do, still need to go towards the suggestion in this thread of exposing a `BadgeLabelStyle` rather than a `Style`

Reply:
Yes, we need to access this in Objective C class, making it optional (since override happening, user can reset too) we need to add wrapper as Enum optional is not compliable.

Not clear with point - `Assuming we do, still need to go towards the suggestion in this thread of exposing a BadgeLabelStyle rather than a Style`, can you please explain bit more.

Reply:
`Style` (exposed to objC with MSFNavigationBarTitleAccessoryStyle) and `BadgeLabelStyle` (exposed to objC with MSFBadgeLabelStyle) are 2 different types, right?

Reply:
BadgeLabelButton is created internally in this class. So we won't have access to that object, ideally main app not aware about this thing.
![Screenshot 2024-03-05 at 10 29 40 AM](https://github.com/microsoft/fluentui-apple/assets/102029000/e203f02a-c0f4-46d3-aa4c-48b84843fed6)

Reply:
Updated code to override `badgeLabelStyle` instead of `Style`. But still we require wrapper to make sure we can set to nil if we don't want override.

Reply:
marking resolve as changes are made